src/System.IO.Compression/src/System/IO/Compression/DeflateStream.cs

```diff
@@ -601,7 +618,16 @@ public override Task WriteAsync(Byte[] array, int offset, int count, Cancellatio
             Interlocked.Increment(ref _asyncOperations);
             try
             {
-                await base.WriteAsync(array, offset, count, cancellationToken).ConfigureAwait(false);
+                EnsureNotDisposed();
+
+                await WriteDeflaterOutputAsync(cancellationToken).ConfigureAwait(false);
+
+                // Pass new bytes through deflater
+                _deflater.SetInput(array, offset, count);
+
+                await WriteDeflaterOutputAsync(cancellationToken).ConfigureAwait(false);
+
+                _wroteBytes = true;
```

Comment:
The first Write should only be executed after input has been set from a previous write. In that case _wrotebytes will have already been set to true. We don't set it to false anywhere.